docs/api-planning.md

```diff
@@ -0,0 +1,58 @@
+
+# Learnings from v3 api
+
+## What could be better about the V3 api
```

Comment:
More documentation on the structure! It was difficult for me to get to grips with how the various packages interacted with each other, especially where there were a lot of empty directories and rabbit holes you could get stuck in for a while. As a result, it was very hard to write tests for changes you want to make as you would often struggle to find the corresponding test file (not helped by the long updater times!)

Reply:
added some points on this